File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    p1 = [1,2,3,4,5]
    p2 = [2,1,2,3,2,4,2,5]
    p3 = [3,3,1,1,2,2,4,4,5,5]
    score = [0, 0, 0]

    for i, ans in enumerate(answers):
        if ans == p1[i%len(p1)]:
            score[0] += 1
        if ans == p2[i%len(p2)]:
            score[1] += 1
        if ans == p3[i%len(p3)]:
            score[2] += 1
    
    for i, res in enumerate(score):
        if (res==max(score)):
            answer.append(i+1)
    
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <map>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    int supo1[] = {1,2,3,4,5};
    int supo2[] = {2,1,2,3,2,4,2,5};
    int supo3[] = {3,3,1,1,2,2,4,4,5,5};
    
    map<int, int> mp;
    vector<pair<int, int>> v;
    
    for(int i=0; i<answers.size(); i++) {
        if (answers[i] == supo1[i%5]) mp[1]++;
        if (answers[i] == supo2[i%8]) mp[2]++;
        if (answers[i] == supo3[i%10]) mp[3]++;
    }
    for(auto a: mp) {
        v.push_back({a.second, a.first});
    }
    sort(v.rbegin(), v.rend());
    for(int i=0; i<v.size(); i++) {
        if (i>0 && v[i].first != v[i-1].first) break;
        answer.push_back(v[i].second);
    }
    sort(answer.begin(), answer.end());
    
    return answer;
}